components/eamxx/src/share/atm_process/IOPDataManager.cpp

```diff
@@ -321,184 +306,62 @@ setup_io_info(const std::string& file_name,
-
-    // Find column index of closest lat/lon to target_lat/lon params
-    auto lat_v = fields[0].get_view<Real*>();
-    auto lon_v = fields[1].get_view<Real*>();
-    const auto target_lat = m_params.get<Real>("target_latitude");
-    const auto target_lon = m_params.get<Real>("target_longitude");
-    using minloc_t = Kokkos::MinLoc<Real,int>;
-    using minloc_value_t = typename minloc_t::value_type;
-    minloc_value_t minloc;
-    Kokkos::parallel_reduce(ncols, KOKKOS_LAMBDA (int icol, minloc_value_t& result) {
-      auto dist = std::abs(lat_v(icol)-target_lat)+std::abs(lon_v(icol)-target_lon);
-      if(dist<result.val) {
-        result.val = dist;
-        result.loc = icol;
-      }
-    }, minloc_t(minloc));
-
-    // Find processor with closest lat/lon match
-    const auto my_rank = m_comm.rank();
-    std::pair<Real, int> min_dist_and_rank = {minloc.val, my_rank};
-    m_comm.all_reduce<std::pair<Real, int>>(&min_dist_and_rank, 1, MPI_MINLOC);
-
-    // Broadcast closest lat/lon values to all ranks
-    const auto lat_v_h = lat_f.get_view<Real*,Host>();
-    const auto lon_v_h = lon_f.get_view<Real*,Host>();
-    auto local_column_idx = minloc.loc;
-    auto min_dist_rank = min_dist_and_rank.second;
-    Real lat_lon_vals[2];
-    if (my_rank == min_dist_rank) {
-      lat_lon_vals[0] = lat_v_h(local_column_idx);
-      lat_lon_vals[1] = lon_v_h(local_column_idx);
-    }
-    m_comm.broadcast(lat_lon_vals, 2, min_dist_rank);
-
-    // Set local_column_idx=-1 for mpi ranks not containing minimum lat/lon distance
-    if (my_rank != min_dist_rank) local_column_idx = -1;
+    // Read lat/lon fields from dile
+    auto lat = grid->create_geometry_data("lat",grid->get_2d_scalar_layout());
+    auto lon = grid->create_geometry_data("lon",grid->get_2d_scalar_layout());
+    AtmosphereInput latlon_reader (file_name,grid,{lat,lon});
+    latlon_reader.read_variables();
 
-    // Store closest lat/lon info for this grid, used later when reading ICs
-    m_lat_lon_info[grid_name] = ClosestLatLonInfo{lat_lon_vals[0], lat_lon_vals[1], min_dist_rank, local_column_idx};
+    m_io_grids[grid_name] = grid;
   }
 }
 
 void IOPDataManager::
 read_fields_from_file_for_iop (const std::string& file_name,
-                               const vos& field_names_nc,
-                               const vos& field_names_eamxx,
-                               const util::TimeStamp& initial_ts,
-                               const field_mgr_ptr field_mgr,
-                               const int time_index)
+                               const std::vector<Field>& fields,
+                               const std::shared_ptr<const AbstractGrid>& grid)
 {
-  const auto dummy_units = ekat::units::Units::nondimensional();
-
-  EKAT_REQUIRE_MSG(field_names_nc.size()==field_names_eamxx.size(),
-                  "Error! Field name arrays must have same size.\n");
-
-  if (field_names_nc.size()==0) {
+  if (fields.size()==0) {
+    // Does this ever happen?
```

Comment:
Probably not? We could remove this check make it error out because it doesn't make sense to call.

Reply:
I can replace it with an assert in the next PR.